src/core/parser_internal/docgen_worker.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "../symbol.hpp"
+#include "../token.hpp"
+#include "../json_writer.hpp"
+#include "../parse_worker.hpp"
+#include "../worker_routine.hpp"
+
+namespace docgen {
+namespace core {
+namespace parser_internal {
+
+using token_class_t = Token<Symbol>;
+
+using writer_class_t = JSONWriter<token_class_t>;
+
+using worker_class_t = ParseWorker<token_class_t, writer_class_t>;
+
+using routine_details_t = WorkerRoutine<worker_class_t>;
+
+} // namespace parser_internal
+
+inline bool operator==(const Token<Symbol>& t1, const Token<Symbol>& t2) {
```

Comment:
This and/or the hash function below may belong in `Token.hpp`

Reply:
yes both of these should be in `token.hpp`. This file does contain the generic `token` but it also contains specialized behavior for `Token<Symbol>`.

Reply:
Okay yes ty, the specialization was what I was concerned about.